packages/react-scripts/scripts/init.js

```diff
@@ -76,6 +82,16 @@ module.exports = function(appPath, appName, verbose, originalDirectory) {
   }
   args.push('react', 'react-dom');
 
+  // Install additional template dependencies, if present
+  var templateDependenciesPath = path.join(appPath, '.template.dependencies.json');
```

Comment:
The main problem is that a `package.json` is created automatically when the npm project is initialized (by `create-react-app`).

If a `package.json` is added to the template, there should be a merge strategy between the generated one and the template one, and the most obvious would be that of merging only *(dev)dependencies*.
This case would be, IMO far more unreliable.

By adding a "special" file name I think it's more obvious that it is not supposed to work like `package.json` at all.

----

The main problem I find in this hole situation is that we're adding the `--template` cli option, that opens ways to create a custom template, but (for now^) should be only for internal usage.